bundles/org.openhab.binding.meater/src/main/java/org/openhab/binding/meater/internal/api/MeaterRestAPI.java

```diff
@@ -0,0 +1,177 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.meater.internal.api;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpStatus;
+import org.openhab.binding.meater.internal.MeaterBridgeConfiguration;
+import org.openhab.binding.meater.internal.MeaterException;
+import org.openhab.binding.meater.internal.dto.MeaterProbeDTO;
+import org.openhab.binding.meater.internal.dto.MeaterProbeDTO.Device;
+import org.openhab.core.i18n.LocaleProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link MeaterRestAPI} class defines the Meater REST API
+ *
+ * @author Jan Gustafsson - Initial contribution
+ */
+@NonNullByDefault
+public class MeaterRestAPI {
+    private static final String API_ENDPOINT = "https://public-api.cloud.meater.com/v1/";
+    private static final String JSON_CONTENT_TYPE = "application/json";
+    private static final String LOGIN = "login";
+    private static final String DEVICES = "devices";
+    private static final int MAX_RETRIES = 3;
+
+    private final Logger logger = LoggerFactory.getLogger(MeaterRestAPI.class);
+    private final Gson gson;
+    private final HttpClient httpClient;
+    private final MeaterBridgeConfiguration configuration;
+    private String authToken = "";
+    private String userId = "";
+    private LocaleProvider localeProvider;
+
+    public MeaterRestAPI(MeaterBridgeConfiguration configuration, Gson gson, HttpClient httpClient,
+            LocaleProvider localeProvider) {
+        this.gson = gson;
+        this.configuration = configuration;
+        this.httpClient = httpClient;
+        this.localeProvider = localeProvider;
+    }
+
+    public boolean refresh(Map<String, MeaterProbeDTO.Device> meaterProbeThings) {
+        try {
+            // Login
+            login();
```

Comment:
If we already have an `authToken`, I don't think it's needed to log in every 30 seconds? See https://github.com/apption-labs/meater-cloud-public-rest-api#api-basics - "The JWT doesn't expire but may be reset." The reset part should be already handled by `getFromApi()` by reacting to HTTP code 401.

Reply:
Fixed!

Reply:
Resolving this comment, please see [other comment](https://github.com/openhab/openhab-addons/pull/13400#discussion_r988354622) with proposal for using `authToken` to determine whether authentication is needed.